Convex_hull_3/test/Convex_hull_3/ch3_test_do_intersect.cpp

```diff
@@ -0,0 +1,261 @@
+        }
+
+    //Test between cubes
+    std::vector<P> cube_bis;
+    for(int x=0; x<2; ++x)
+      for(int y=0; y<2; ++y)
+        for(int z=0; z<2; ++z)
+            cube_bis.push_back(P(x,y,z));
+
+    auto transform=[](std::vector<P> &cube, const CGAL::Aff_transformation_3<K> &t){
+      for(auto &p: cube)
+        p=t(p);
+    };
+    //Test their intersection for many translations
+    for(double x=-1.5; x<=1.5; x+=0.5)
+      for(double y=-1.5; y<=1.5; y+=0.5)
+        for(double z=-1.5; z<=1.5; z+=0.5){
+          CGAL::Aff_transformation_3<K> t(CGAL::TRANSLATION, V(x,y,z));
+          transform(cube_bis, t);
+          test(cube, cube_bis, std::abs(x)<1.5 && std::abs(y)<1.5 && std::abs(z)<1.5);
+          transform(cube_bis, t.inverse());
+        }
+  }
+
+  void test_degenerate()
+  {
+    //Vertices
+    std::vector<P> origin(1, CGAL::ORIGIN);
+    std::vector<P> vertex1(1, P(1,0,0));
+    std::vector<P> vertex2(1, P(0,1,0));
+
+    test(origin, origin, true);
+    test(vertex1,vertex1, true);
+
+    test(origin, vertex1, false);
+    test(vertex1, vertex2, false);
+
+    //Segments
+    std::vector<P> seg1({P(0,0,0),P(2,0,0)});
+    std::vector<P> seg2({P(0,0,0),P(0,2,0)});
+    std::vector<P> seg3({P(0,2,0),P(2,2,0)});
+    std::vector<P> seg4({P(1,1,0),P(1,3,0)});
+
+    test(origin, seg1, true);
+    test(vertex1, seg1, true);
+
+    test(vertex2, seg1, false);
+
+    test(seg1, seg1, true);
+    test(seg1, seg2, true);
+    test(seg3, seg4, true);
+
+    test(seg1, seg3, false);
+    test(seg1, seg4, false);
+
+    //Triangle
+    std::vector<P> tr1({P(0,0,0),P(2,0,0),P(0,2,0)});
+    std::vector<P> tr2({P(0,0,0),P(2,0,0),P(0,0,2)});
+    std::vector<P> tr3({P(0,0,0),P(2,0,2),P(0,0,2)});
+    std::vector<P> tr4({P(1,0,0),P(2,0,2),P(0,0,2)});
+    std::vector<P> tr5({P(0,0,2),P(2,0,2),P(0,2,2)});
+
+    test(tr1, tr1, true);
+    test(tr1, tr2, true);
+    test(tr1, tr3, true);
+    test(tr1, tr4, true);
+    test(tr3, tr4, true);
+
+    test(tr1, tr5, false);
+    test(tr5, tr1, false);
+  }
+
+  void test_half_sphere()
+  {
+    std::vector<P> half_sphere;
+    // constexpr K::FT eps(std::pow(2,-40));
+    constexpr double pi=3.14159265358979323846;
+    for(double phi=25./16.; phi>0; phi-=1./4.)
+      for(double theta=0; theta<2*pi; theta+=0.25)
+            half_sphere.push_back(P(std::sin(phi) * std::cos(theta),
+                                    std::sin(phi) * std::sin(theta),
+                                    std::cos(phi)));
+
+    for(double x=-1; x<=1; x+=0.1)
+      for(double y=-1; y<=1; y+=0.1){
+        std::vector<P> outside(1, P(x,y,std::nextafter(std::cos(25./16),0)));
+        std::vector<P> inside(1, P(x,y,std::nextafter(std::cos(25./16),1)));
+
+        test(half_sphere, inside, x*x+y*y<0.99);
+        test(half_sphere, outside, false);
+      }
+  }
+
+  void test_random_tetrahedrons(int N, CGAL::Random &r)
+  {
+    using Tet = typename K::Tetrahedron_3;
+
+    auto random_point=[&](){
```

Comment:
Use the generators from [this package](https://doc.cgal.org/latest/Generator/index.html#Chapter_Geometric_Object_Generators).